Invalid Color initialization not detected on some platforms

Using an int value that is out of the expected range of 0x0 to 0xFFFFFFFF is not detected as an error on some platforms.

e.g. `color = pygame.Color(0x1FFFFFFFF)` should raise a `ValueError`.

This issue was seen during the [travis-ci](https://travis-ci.org/pygame/pygame/builds/534526560) for PR #1058.

```
FAIL: test_color__int_arg_invalid (pygame.tests.color_test.ColorTypeTest)
Ensures invalid int values are detected when creating Color objects.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/pygame-2.0.0.dev1-py3.7-macosx-10.11-x86_64.egg/pygame/tests/color_test.py", line 406, in test_color__int_arg_invalid
    color = pygame.Color(0x1FFFFFFFF)
AssertionError: ValueError not raised
```

--- a/src_c/color.c
+++ b/src_c/color.c
@@ -390,7 +390,7 @@ _get_color(PyObject *val, Uint32 *color)
 #if !PY3
     if (PyInt_Check(val)) {
         long intval = PyInt_AsLong(val);
-        if (intval == -1 && PyErr_Occurred()) {
+        if ((intval == -1 && PyErr_Occurred()) || (intval > 0xFFFFFFFF)) {
             PyErr_SetString(PyExc_ValueError, "invalid color argument");
             return 0;
         }
@@ -400,7 +400,7 @@ _get_color(PyObject *val, Uint32 *color)
 #endif
     if (PyLong_Check(val)) {
         unsigned long longval = PyLong_AsUnsignedLong(val);
-        if (PyErr_Occurred()) {
+        if (PyErr_Occurred() || (longval > 0xFFFFFFFF)) {
             PyErr_SetString(PyExc_ValueError, "invalid color argument");
             return 0;
         }

--- a/test/color_test.py
+++ b/test/color_test.py
@@ -399,8 +399,6 @@ class ColorTypeTest (unittest.TestCase):
             self.assertEqual(color.b, (value >> 8) & 0xFF)
             self.assertEqual(color.a, value & 0xFF)
 
-    # The skip decorator can be removed when issue #1060 is resolved.
-    @unittest.skip('invalid int value not detected on some platforms')
     def test_color__int_arg_invalid(self):
         """Ensures invalid int values are detected when creating Color objects.
         """

